### PR TITLE
Static adapter

### PR DESCRIPTION
**Pull request by xFloooo**
Created to move without history. This is an opportunity Server Side Rendering, see example https://github.com/xFloooo/mobx-state-router-ssr-boilerplate

### CODE REVIEW

**Comment by nareshbhatia** [comment]
Thanks @xFloooo for the PR. Will be looking at this today.

**Comment by xFloooo** [comment]
Example -
 https://github.com/xFloooo/mobx-state-router-ssr-boilerplate

**Comment by nareshbhatia** [comment]
The example doesn't work for me. When I go to http://localhost:5555/, the server console displays the following errors and the browser does not display anything:

> StaticAdapter.goToLocation("/")
RouterStore.transition({"routeName":"__initial__","params":{},"queryParams":{}}, {"routeName":"home","params":{},"queryParams":{}})
StaticAdapter.preloadReady()
RouterView.render() - {"routeName":"home","params":{},"queryParams":{}}
(node:90207) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: path_to_regexp_1.compile is not a function
(node:90207) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by xFloooo** [comment]
Forgot to add dependencies. Fix example, now need `cd mobx-state-router && npm i`

**Comment by nareshbhatia** [comment]
@xFloooo, I reviewed the PR today. I think the base concepts are solid, however there are places where we can clean up and simplify the API. I will start commenting on the changes I would like to see. Also I created a branch called `ssr` - let's merge to this branch first, instead of master.

**Comment by xFloooo** [comment]
@nareshbhatia, do I close this PR and create a new one in the SSR branch?

**Comment by nareshbhatia** [comment]
Yes, please create a new PR.

File: src/adapters/static-adapter.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    goToLocation(location: string, search: string): Promise<any> {
+        if (process.env.NODE_ENV === 'development') {
+            console.log(
+                `StaticAdapter.goToLocation(${JSON.stringify(
+                    `${this.fullLocation}`
+                )})`
+            );
+        }
+
+        const routes = this.routerStore.routes;
```
[the remark this answers is not in the document]
Comment (xFloooo) [reply src/adapters/static-adapter.ts]: If we want to return a `Promise<RouterState>`, then we need the else branch, on this these lines and differ

fail

[the remark this answers is not in the document]
Comment (xFloooo) [reply src/adapters/static-adapter.ts]: You can make changes to the `master`, and I will do a `rebase`

**Pull request closed without merging**